src/agreement.rs

```diff
@@ -98,6 +98,7 @@ derive_debug_from_field!(Algorithm, i);
 /// An ephemeral private key for use (only) with `agree_ephemeral`. The
 /// signature of `agree_ephemeral` ensures that an `EphemeralPrivateKey` can be
 /// used for at most one key agreement.
+#[derive(Clone, Copy)]
```

Comment:
This part is definitely incompatible with the design for `EphemeralPrivateKey`.  `EphemeralPrivateKey` is intentionally move-only so that we can use the type system to ensure that the key is only used once.